### PR TITLE
Created a UserService Test

### PR DESCRIPTION
**Pull request by dapariciom**
Created a UserService Test

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,6 +76,12 @@
 			<artifactId>jjwt</artifactId>
 			<version>0.9.1</version>
 		</dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>4.8.0</version>
+            <scope>compile</scope>
```
Comment (BeeAM17) [inline pom.xml]: Can we remove it?

Comment (dapariciom) [reply pom.xml]: Yes. Let me update it

File: src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Test
+    void testSignUpWithNullUser(){
+
+        try{
+            authController.signUp(null);
+        } catch (ResponseStatusException e){
+            assertEquals("User request is missing or empty", e.getReason());
+            assertEquals(HttpStatus.BAD_REQUEST, e.getStatus());
+        }
```
Comment (BeeAM17) [inline src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java]: Use AssertThrows here

https://howtodoinjava.com/junit5/expected-exception-example/

File: src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                .roles(ImmutableSet.of(roleEntity))
+                .build();
+
+        when(userService.findByUserName(userRequest.getUserName())).thenReturn(Optional.of(userToReturn));
+
+        try{
+            authController.signUp(userRequest);
+        } catch (ResponseStatusException e){
+            assertEquals("Username already exists", e.getReason());
+            assertEquals(HttpStatus.BAD_REQUEST, e.getStatus());
```
Comment (BeeAM17) [inline src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java]: Same here

File: src/test/java/com/example/shorturl/service/user/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void testFindUserByName(){
+
+        String findName = "admin";
+
+        userService.findByUserName(findName);
+
+        verify(userRepository, times(1)).findByUserName(any());
+
```
Comment (BeeAM17) [inline src/test/java/com/example/shorturl/service/user/UserServiceTest.java:58]: I don't see any test if email is not valid or username is null. What will be the behaviour of the code?

**Review by BeeAM17 — COMMENTED** [review]
Few changes remaining

File: src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java (hunk 2 above)
Comment (dapariciom) [reply src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java]: Thanks. Fixed :)

File: src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java (hunk 3 above)
Comment (dapariciom) [reply src/test/java/com/example/shorturl/controller/login/AuthControllerTest.java]: Fixed

File: src/test/java/com/example/shorturl/service/user/UserServiceTest.java (hunk 4 above)
Comment (dapariciom) [reply src/test/java/com/example/shorturl/service/user/UserServiceTest.java:58]: Let me add it

Comment (dapariciom) [reply src/test/java/com/example/shorturl/service/user/UserServiceTest.java:58]: Done

**Review by BeeAM17 — APPROVED** [review]
Nice work!

**Pull request merged**